File: pidroid/src/main/java/com/suaro/pidroid/core/NativeMethods.java
package com.suaro.pidroid.core;

import android.content.res.AssetManager;

import org.jetbrains.annotations.NotNull;


public final class NativeMethods {
    static {
        System.loadLibrary("pidroid");
    }
    public static native void setup(@NotNull PidroidConfig var1, @NotNull AssetManager var2);

    public static native void detectFace(@NotNull int[] var1, int var2, int var3, @NotNull FaceDetectionResult var4);
}

File: pidroid/src/main/java/com/suaro/pidroid/core/NativeMethods.kt
package com.suaro.pidroid.core

import android.content.res.AssetManager


class NativeMethods {

    companion object {
        init {
            System.loadLibrary("pidroid")
        }

        external fun setup(config: PidroidConfig, mgr: AssetManager)
        external fun detectFace(rgbaBytes: IntArray, width: Int, height: Int, detectionInfo: FaceDetectionResult)
    }
}